File: Modules/ServerTypes/hospital/DoctorsForm.ts
import { StringEditor, LookupEditor, EnumEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { TypeEnum } from "../Modules/hospital.Patients.Enum.TypeEnum";

export interface DoctorsForm {
    DoctorName: StringEditor;
    SpecialityList: LookupEditor;
    Type: EnumEditor;
}

export class DoctorsForm extends PrefixedContext {
    static readonly formKey = 'hospital.Doctors';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!DoctorsForm.init)  {
            DoctorsForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;
            var w2 = EnumEditor;

            initFormType(DoctorsForm, [
                'DoctorName', w0,
                'SpecialityList', w1,
                'Type', w2
            ]);
        }
    }
}

[TypeEnum]; // referenced types

File: Modules/hospital/Doctors/DoctorsForm.cs
namespace Serene2.hospital.Forms;

using Serene2.Modules.hospital.Patients.Enum;
using Serenity.ComponentModel;


[FormScript("hospital.Doctors")]
[BasedOnRow(typeof(DoctorsRow), CheckNames = true)]
public class DoctorsForm
{
        public string DoctorName { get; set; }
        
        public List<int> SpecialityList { get; set; }

        public TypeEnum Type { get; set; }
}
 